src/v/raft/consensus.cc

```diff
@@ -3959,7 +3959,7 @@ reply_result consensus::lightweight_heartbeat(
           target_node,
           _self,
           source_node);
-        return reply_result::failure;
```

Comment:
there is a logic in heartbeat_manager that uses a failure to trigger the full heratbeat, it seems that in this case it won't do it.

Reply:
Do we need a full heartbeat if this is the wrong node. It seems to me we have to find the correct node first :)